### PR TITLE
Feature/exhaustive matching

### PR DESCRIPTION
**Pull request by wpalmeri**
*  :weary: :briefcase: matching
*  more logical section ordering

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -400,6 +398,13 @@ Option(123) match {
 }
 ```
 
+## Case Ordering
```
Comment (taylorleese) [inline README.md]: @wpalmeri @ayakushev99 Sounds like you need to talk. https://github.com/paypal/scala-style-guide/pull/9

File: README.md [hunk 2]
```diff
@@ -377,17 +377,15 @@ In general, logic that handles a choice between two or more outcomes should pref
 
 When you `match` on any type, follow these rules:
 
-1. indent all `case` statements at the same level, and put the `=>` one space to
+1. Pattern matching should be exhaustive and explicitly handle the failure/default case rather than relying on a runtime `MatchError`. Case classes used in pattern matching should extend a common `sealed trait` so that compiler warnings will be generated for inexhaustive matching.
```
Comment (taylorleese) [inline README.md]: @wpalmeri Partial functions are a bit of an exception here. It might be worth calling that out.

**Comment by wpalmeri** [comment]
@taylorleese check the first bullet again. also merged from develop.

**Comment by taylorleese** [comment]
:shipit:

**Pull request merged**